### ISSUE TITLE
Hunner/Hiera module - Forge or GitHub?

### ISSUE THREAD

**Issue by roberteckdale** [puppetlabs/control-repo#17]
Pretty much a complete newb so I apologize in advance if this is a stupid question:

The control-repo Puppetfile includes Hunner's 'hiera' module from GitHub but it looks like the same module is also available on the Forge (v2.0.0 versus GitHub's 1.4.1). Any reason not to use the module available on the Forge?

Secondary question: Is this even the 'correct' forum for such a question. Kinda feels like it's not.

**Comment by roberteckdale** [comment]
Closing this as it's pretty apparent to me that:
1. This isn't the correct forum
2. My ignorance shouldn't cost you time.

Apologies for the noise.

**Comment by npwalker** [comment]
@roberteckdale 2.0.0 was released yesterday.  I intend to use it someday but I'll have to probably change some code and re-test so I'm not in a hurry.  The version we have here achieves what we need for this repo.  

As for forge vs github, there's no particular reason.  I wanted to provide examples of different ways to use the Puppetfile.  

It might not be the correct forum but that's not a big deal.  If you'd like to have a discussion about this type of thing I would suggest 2 possible places.  

1. ask.puppetlabs.com 
2. pe-users  ( https://groups.google.com/a/puppetlabs.com/forum/#!forum/pe-users )

**Comment by glarizza** [comment]
@roberteckdale I'll respond since it's here :)

We tend to choose Github versus the Forge RIGHT NOW because the underlying way that R10k/Code Manager fetches data from Github is via git....so if there's an update that needs to be made (i.e. the version on your server is older than specified) it uses git and its ability to pull only the delta (i.e. changes) between the version on your server and what's up on Github.

If you use the Forge and it notices that you're using a different version of the module in question, it pulls down the entire module.  That uses a bit more bandwidth.  Also, if you use a git-based source (i.e. github or your own internal repo), R10k/Code Manager maintain a git cache locally on disk...so if you CAN'T reach github it will still use that cache (and then update the cache when it can).  

In summary, it's a bit faster/smarter as of right now.  We're trying to make that better when seeking from the Forge, but right now git wins :)

**Comment by roberteckdale** [comment]
Thanks for the informative responses @npwalker and @glarizza.
Much appreciated.
